frontend/app/src/components/calendar/CalendarReminder.vue

```diff
@@ -0,0 +1,313 @@
+<script setup lang="ts">
+import type { CalendarEvent } from '@/types/history/calendar';
+import type { CalendarReminderTemporaryPayload, CalenderReminderPayload } from '@/types/history/calendar/reminder';
+
+const props = withDefaults(
+  defineProps<{
+    editableItem?: CalendarEvent;
+  }>(),
+  {
+    editableItem: undefined,
+  },
+);
+const { editableItem } = toRefs(props);
+
+const { t } = useI18n();
+
+const showReminders: Ref<boolean> = ref(false);
+
+const temporaryData: Ref<CalendarReminderTemporaryPayload[]> = ref([]);
+
+const length = computed(() => get(temporaryData).filter(item => item.secsBefore > 0).length);
+
+const {
+  fetchCalendarReminders,
+  addCalendarReminder,
+  editCalendarReminder,
+  deleteCalendarReminder,
+} = useCalendarReminderApi();
+
+const { notify } = useNotificationsStore();
+
+async function addCalendarReminderHandler(reminders: CalenderReminderPayload[]) {
+  try {
+    const result = await addCalendarReminder(reminders);
+    if (result.failed && result.failed.length > 0) {
+      notify({
+        display: true,
+        title: t('calendar.reminder.add_error.title'),
+        message: t('calendar.reminder.add_error.some_failed', {
+          ids: result.failed.join(', '),
+        }),
+      });
+    }
+  }
+  catch (error: any) {
+    logger.error(error);
+    notify({
+      display: true,
+      title: t('calendar.reminder.add_error.title'),
+      message: t('calendar.reminder.add_error.message', {
+        message: error.message,
+      }),
+    });
+  }
+}
+
+async function refreshTemporaryData() {
+  const item = get(editableItem);
+  if (item) {
```

Comment:
Any chance we can inverse and return if the item is not defined to avoid extra nesting?